airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/offset_increment.py

```diff
@@ -36,16 +37,23 @@ class OffsetIncrement(PaginationStrategy):
     """
 
     config: Config
-    page_size: Union[str, int]
+    page_size: Optional[Union[str, int]]
     parameters: InitVar[Mapping[str, Any]]
+    decoder: Decoder = JsonDecoder(parameters={})
 
     def __post_init__(self, parameters: Mapping[str, Any]):
         self._offset = 0
         page_size = str(self.page_size) if isinstance(self.page_size, int) else self.page_size
-        self._page_size = InterpolatedString(page_size, parameters=parameters)
+        if page_size:
+            self._page_size = InterpolatedString(page_size, parameters=parameters)
+        else:
+            self._page_size = None
 
     def next_page_token(self, response: requests.Response, last_records: List[Mapping[str, Any]]) -> Optional[Any]:
-        if len(last_records) < self._page_size.eval(self.config):
+        decoded_response = self.decoder.decode(response)
+
+        # Stop paginating when there are fewer records than the page size or the current page has no records
+        if (self._page_size and len(last_records) < self._page_size.eval(self.config, response=decoded_response)) or len(last_records) == 0:
```

Comment:
Thinking about the use case of APIs that do not allow specifying a `page_size`, there is actually one particular use case that we need to also consider. The case is when the API response contains paging metadata that we need to use to determine if there are any more records. Although the number of records to fetch is fixed and cannot be included in the request, we still need to reference a page_size limit in the response to stop paging. Which is why for the Offset strategy, we’re also decoding the response and allowing it to be part of the interpolation context.

See this example in Discourse, where `meta.limit` is needed to stop pagination:
https://docs.discourse.org/#tag/Groups/operation/listGroupMembers